botocore/vendored/requests/api.py

```diff
@@ -45,6 +55,10 @@ def request(method, url, **kwargs):
       >>> req = requests.request('GET', 'http://httpbin.org/get')
       <Response [200]>
     """
+    warnings.warn(
+        _WARNING_MSG.format(name=method),
+        PendingDeprecationWarning
```

Comment:
This should be `DeprecationWarning` since it's already deprecated. Though I'm not sure deprecated is right at all since this was never meant to be used in the first place. Might be better to stick with `UserWarning` since that will be shown by default.

Reply:
Hmm, I think either `PendingDeprecationWarning` or `DeprecationWarning` is appropriate.
I'm -1 on labeling this as a `UserWarning` just to get it to show up by default, as I think it would go against what users would expect of a deprecation warning.

Reply:
Let's switch it to `DeprecationWarning` for now.  I think we'll eventually convert this to a warning that shows up by default, but I want to put something in place now and let it sit for a while before we start printing warnings to stderr.